supervisor/supervisor.go

```diff
@@ -48,9 +48,6 @@ func New(stateDir string, runtimeName, shimName string, runtimeArgs []string, ti
 	}
 	go s.exitHandler()
 	go s.oomHandler()
-	if err := s.restore(); err != nil {
```

Comment:
> restore gets deadlocked

Before this pr, 
RegisterHealthServer -> grpc Serve -> restore -> RegisterAPIServer -> handtask

After this pr,
RegisterHealthServer -> RegisterAPIServer -> grpc Serve -> restore -> handtask

So if the issue exist , then I think the issue is already exist before changing the order, on the other hand , the issue should not exist, if exist, it is fatal, we must fix it. 

If we still worry about it, we can have a new issue to track it(restore gets deadlocked).